prog/vm/github_runner.rb

```diff
@@ -95,6 +95,15 @@ def before_run
     # https://github.com/microsoft/azure-pipelines-agent/issues/3461
     vm.sshable.cmd("echo \"PATH=$PATH\" >> .env")
 
+    # Docker containers can't resolve DNS addresses by default on our networking
+    # setup. We will investigate it in depth, and try to find more generic solution.
+    # Related issue: https://github.com/ubicloud/ubicloud/issues/507
+    # Until proper fix, we add custom systemd-resolved configuration.
+    # Docker gets resolve.conf content from systemd-resolved service.
+    vm.sshable.cmd("sudo mkdir -p /etc/systemd/resolved.conf.d")
+    vm.sshable.cmd("sudo sh -c 'echo \"[Resolve]\nDNS=9.9.9.9\" > /etc/systemd/resolved.conf.d/Ubicloud.conf'")
```

Comment:
Not sure how long this code is for this world, but it wouldn't hurt too much to add the redundant DNS servers.

```suggestion
    vm.sshable.cmd("sudo sh -c 'echo \"[Resolve]\nDNS=9.9.9.9 149.112.112.112 2620:fe::fe 2620:fe::9\" > /etc/systemd/resolved.conf.d/Ubicloud.conf'")
```

Space-separated list syntax demonstrated here: https://wiki.archlinux.org/title/Systemd-resolved